src/services.py

```diff
@@ -0,0 +1,153 @@
+        font = ImageFont.truetype(self.fontDirectory, 50)
+        draw.text((150, y_text+100), "START : ",
+                  fill="white", font=font, align="left")
+        timeDifference = contest.endDateTime - contest.startDateTime
+        contest.startDateTime = str(
+            contest.startDateTime.strftime('%d-%m-%Y %H:%M:%S'))
+        contest.endDateTime = str(
+            contest.endDateTime.strftime('%d-%m-%Y %H:%M:%S'))
+        draw.text((400, y_text+100), contest.startDateTime,
+                  fill="white", font=font, align="left")
+        draw.text((150, y_text+200), "END : ", fill="white",
+                  font=font, align="right")
+        draw.text((400, y_text+200), contest.endDateTime,
+                  fill="white", font=font, align="right")
+        timeDifference = str(timeDifference)
+        duration = ""
+        print(timeDifference)
+        durationList = timeDifference
+        if "days" in timeDifference:
+            daysList = timeDifference.split(" days, ")
+            duration = daysList[0] + " days "
+        if "days" in timeDifference:
+            durationList = timeDifference.split(" days, ")
+            durationList = durationList[1]
+        #print(durationList)
+        durationList = durationList.split(":")
+        print("hi")
+        if(durationList[0] != "00"):
+            duration += str(durationList[0]) + " Hours "
+        if(durationList[1] != "00"):
+            duration += str(durationList[1]) + " Minutes "
+        if(durationList[2] != "00"):
+            duration += str(durationList[2]) + " Seconds"
+        w, h = font.getsize("DURATION : " + duration)
+        draw.text((((MAX_W - w) / 2), y_text+300), "DURATION : " +
+                  str(duration), fill="white", font=font, align="right")
+        return image
+class ContestsRetreiver:
+    def getTodaysContestDetails(self):
+        # 2020-05-11 15:00:00
+        contestsJson = []
+        codechefContests = self.getContestsfromCodechef()
+        for contest in codechefContests:
+            if(datetime.now().date() == parser.parse(contest['start_time']).date()):
+                contestsJson.append(contest)
+        codeforcesContests = self.getContestsfromCodeforces()
+        for contest in codeforcesContests:
+            if(datetime.now().date() == parser.parse(contest['start_time']).date()):
+                contestsJson.append(contest)
+
+        return contestsJson
+
+    def getAllUpcomingContestDetails(self):
+        contestsJson = []
+        codechefContests = self.getContestsfromCodechef()
+        for contest in codechefContests:
+            contestsJson.append(contest)
+        codeforcesContests = self.getContestsfromCodeforces()
+        for contest in codeforcesContests:
+            contestsJson.append(contest)
+
+        return contestsJson
+
+    def getContestsfromCodechef(self):
+        URL = "https://www.codechef.com/contests"
+        soup = bs(req.get(URL).content, 'lxml')
```

Comment:
Lets' say the URL doesn't allow the crawler to pass through, then it will throw an error. Is there a way to handle it?